### PR TITLE
Improve resiliency to auto-formatting in server

### PR DESCRIPTION
**Pull request by pugnascotia**
Make a number of changes so that code in the `server` directory is more resilient to automatic formatting. This covers:

* Reformatting multiline JSON to embed whitespace in the strings
* Move some comments around to they aren't auto-formatted to a strange
  place. This also required moving some `&&` and `||` operators from the
  end-of-line to start-of-line`.
* ~Turn of formatting of data tables in `HyperLogLogPlusPlus`. This also
  required a checkstyle suppression.~
* Exclude `HyperLogLogPlusPlus` from being formatted. It would format OK if the rules for wrapping array contents were set to "wrap when necessary", but that would be worse for the rest of the codebase. Instead, I've applied to formatter to the rest of the file, for consistency, but left the data tables alone.
* Add helper method `reformatJson()`, to strip whitespace from a JSON
  document using XContent methods. This is sometimes necessary where
  a test is comparing some machine-generated JSON with an expected
  value.

### CODE REVIEW

**Comment by pugnascotia** [comment]
I've moved the JSON-related changes in this PR to #48553 in order to reduce the number of changes here.

File: buildSrc/src/main/resources/checkstyle_suppressions.xml
Comment (atorok) [inline buildSrc/src/main/resources/checkstyle_suppressions.xml]: I think that's an important check for the documentation I don't think we should be turning it off

**Review by alpar-t — COMMENTED** [review]
LGTM in general do have once concern about check style

File: buildSrc/src/main/resources/checkstyle_suppressions.xml
Comment (pugnascotia) [reply buildSrc/src/main/resources/checkstyle_suppressions.xml]: `HyperLogLogPlusPlus.java` doesn't have any existing doc snippets, so right now it's safe to turn off the check for this file only. I suppose I could have reformatted the tagged sections to meet the width limit, but I didn't want to mess around the file in that way. I could add some comments in the file to explain the situation about formatting in greater detail? What do you think?

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/1

File: buildSrc/src/main/resources/checkstyle_suppressions.xml
Comment (rjernst) [reply buildSrc/src/main/resources/checkstyle_suppressions.xml]: This seems like a hack. I don't think we should mark something as a doc snippet if it isn't intended to be used in the docs. What about that file does not auto format well? Is there anything we can change about the format rules to make it better?

**Comment by pugnascotia** [comment]
Still looking for an approval on this one.

File: buildSrc/src/main/resources/checkstyle_suppressions.xml
Comment (pugnascotia) [reply buildSrc/src/main/resources/checkstyle_suppressions.xml]: I think "hack" is a fair assessment. However, the Eclipse formatter doesn't have support for recognising tables of data (it's the sole feature from the Google formatter than I miss - at least, I'm sure I read that somewhere), so the arrays of data get spread over many, many, many lines. Now, you could argue that this is irrelevant - semantically it makes no difference - but I didn't feel comfortable making that call.

You also can't define multiple tags for disabling and enabling the formatter, and we need to be able to use them for legitimate docs snippets.

The other option is to alter the source to build up the data arrays in pieces and then combine them. This is altering the source to please the compiler, which isn't great. Or define the data in text files or some such, and parse them, which is also a poor option.

...actually, thinking about it again, I suppose we could change the formatter config to only wrap where necessary in array initializers. That would be perfect for `HyperLogLogPlusPlus.java`, but probably worsen readability for other arrays in the codebase. We'd then be configuring formatter for the exceptional cases.

Comment (pugnascotia) [reply buildSrc/src/main/resources/checkstyle_suppressions.xml]: @rjernst I've taken a new approach to `HyperLogLogPlusPlus`, by excluding it from being formatted automatically. I've applied to formatter to the rest of the file, for consistency, but left the data tables alone.

**Comment by pugnascotia** [comment]
@elasticmachine run elasticsearch-ci/2

**Comment by pugnascotia** [comment]
Any more review comments on this?

File: build.gradle
Comment (atorok) [inline build.gradle]: Would be better to just add this to `server/build.gradle`

**Review by alpar-t — APPROVED** [review]
I added a minor comment, no need for another review.

File: build.gradle
Comment (pugnascotia) [reply build.gradle]: I didn't think that would work, but turns out it does, which is much tidier, thank you. Unfortunately, I then had to wrap the config in a try/catch because unless a project opts-in to formatting, the `spotless()` method doesn't exist. I can removed the extra cruft once we enable formatting.

**Pull request merged**
